File: src/Services/StudentService/StudentService.Domain/HistorySemester.cs
using System.ComponentModel;

namespace StudentService.Domain;

public class HistorySemester
{
    public string SemesterCode { get; set; }
    public string SemesterName { get; set; }
    
    [Description("Thời gian bắt đầu kì học")]
    public DateTime EducationStartDate { get; set; }
    
    [Description("Thời gian kết thúc kì học")]
    public DateTime EducationEndDate { get; set; }
    
    [Description("Thời gian kết thúc kì học")]
    public DateTime StartDate { get; set; }
    [Description("Thời gian kết thúc kì học")]
    public DateTime EndDate { get; set; }
    [Description("Lịch sử học")]

    public ICollection<SubjectResult> SubjectResults { get; set; }
}

File: src/Webs/daotao/src/domain/history_semester.ts
import { SubjectResult } from "./subject_result.ts";

export interface HistorySemester {
    semesterCode: string;
    semesterName: string;
    educationStartDate: string;
    educationEndDate: string;
    startDate: string;
    endDate: string;
    subjectResults: SubjectResult[];
}
